File: backend/app/api/engines.py
from __future__ import annotations
from fastapi import APIRouter, Depends, HTTPException
from app.core.interfaces.engine_registry import AbstractEngineRegistry
from app.models.engine_model import EngineInfo

router = APIRouter(prefix="/api/engines", tags=["engines"])


def _to_info(engine) -> EngineInfo:
    return EngineInfo(
        engine_id       = engine.engine_id,
        name            = engine.name,
        description     = engine.description,
        objective       = engine.objective,
        target_map_type = engine.target_map_type.value,
        default_config  = engine.get_default_config(),
        config_schema   = engine.get_config_schema(),
    )


@router.get("", response_model=list[EngineInfo])
def list_engines(registry: AbstractEngineRegistry = Depends(AbstractEngineRegistry)):
    return [_to_info(e) for e in registry.list_all()]


@router.get("/{engine_id}", response_model=EngineInfo)
def get_engine(engine_id: str, registry: AbstractEngineRegistry = Depends(AbstractEngineRegistry)):
    try:
        return _to_info(registry.get(engine_id))
    except KeyError:
        raise HTTPException(status_code=404, detail=f"Engine '{engine_id}' não encontrado")

File: frontend/src/api/engines.ts
import { apiFetch } from './client'
import type { EngineInfo, JSONSchema } from '@/types/engine'

interface RawEngineInfo {
  engine_id:       string
  name:            string
  description:     string
  objective:       string
  target_map_type: string
  default_config:  Record<string, unknown>
  config_schema:   JSONSchema
}

function mapEngine(r: RawEngineInfo): EngineInfo {
  return {
    engineId:      r.engine_id,
    name:          r.name,
    description:   r.description,
    objective:     r.objective,
    targetMapType: r.target_map_type as EngineInfo['targetMapType'],
    defaultConfig: r.default_config,
    configSchema:  r.config_schema,
  }
}

export async function listEngines(): Promise<EngineInfo[]> {
  const raw = await apiFetch<RawEngineInfo[]>('/api/engines')
  return raw.map(mapEngine)
}

export async function getEngine(engineId: string): Promise<EngineInfo> {
  const raw = await apiFetch<RawEngineInfo>(`/api/engines/${engineId}`)
  return mapEngine(raw)
}
